arangod/Aql/ClusterBlocks.cpp

```diff
@@ -1628,3 +1631,305 @@ bool SortingGatherBlock::getBlock(size_t i, size_t atMost) {
+  TRI_ASSERT(result.ok());
+  VPackSlice outDocument = VPackSlice::noneSlice();
+  if(result.buffer){
+    outDocument = result.slice().resolveExternal();
+  }
+
+  VPackSlice oldDocument = VPackSlice::noneSlice();
+  VPackSlice newDocument = VPackSlice::noneSlice();
+  if(outDocument.isObject()) {
+    if(outDocument.hasKey("old")){
+      oldDocument = outDocument.get("old");
+    }
+    if(outDocument.hasKey("new")){
+      newDocument = outDocument.get("new");
+    }
+  }
+
+  // place documents as in the outi variable slots of the result
+  bool aqlValueSet = false;
+  if(out) {
+    if(!outDocument.isNone()){
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(outRegId), AqlValue(outDocument));
+    } else {
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(outRegId), VPackSlice::nullSlice());
+    }
+    aqlValueSet = true;
+  }
+  if(OLD) {
+    TRI_ASSERT(opOptions.returnOld);
+    if(!oldDocument.isNone()){
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(oldRegId), AqlValue(oldDocument));
+    } else {
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(oldRegId), VPackSlice::nullSlice());
+    }
+    aqlValueSet = true;
+  }
+  if(NEW) {
+    TRI_ASSERT(opOptions.returnNew);
+    if(!newDocument.isNone()){
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(newRegId), AqlValue(newDocument));
+    } else {
+      aqlres->emplaceValue(outputCounter, static_cast<arangodb::aql::RegisterId>(newRegId), VPackSlice::nullSlice());
+    }
+    aqlValueSet = true;
+  }
+  traceGetSomeEnd(aqlres);
+  throwIfKilled();  // check if we were aborted
+
+  TRI_IF_FAILURE("SingleRemoteOperationBlock::moreDocuments") {
+    THROW_ARANGO_EXCEPTION(TRI_ERROR_DEBUG);
+  }
+  if(!aqlValueSet) {
+    traceGetSomeEnd(nullptr);
+    return false;
+  }
+  return true;
+}
+
+/// @brief getSome
+AqlItemBlock* SingleRemoteOperationBlock::getSome(size_t atMost) {
+  DEBUG_BEGIN_BLOCK();
+  traceGetSomeBegin(atMost);
+
+  if (_done) {
+    traceGetSomeEnd(nullptr);
+    return nullptr;
+  }
+
+  std::unique_ptr<AqlItemBlock> aqlres;
+
+  RegisterId nrRegs =
+    getPlanNode()->getRegisterPlan()->nrRegs[getPlanNode()->getDepth()];
+  aqlres.reset(requestBlock(atMost, nrRegs));
+
+  int outputCounter = 0;
+  if (_buffer.empty()) {
+    size_t toFetch = (std::min)(DefaultBatchSize(), atMost);
+    if (!ExecutionBlock::getBlock(toFetch)) {
+      _done = true;
+      traceGetSomeEnd(nullptr);
+      return nullptr;
+    }
+    _pos = 0;  // this is in the first block
+  }
+
+  // If we get here, we do have _buffer.front()
+  arangodb::aql::AqlItemBlock* cur = _buffer.front();
+  TRI_ASSERT(cur != nullptr);
+  size_t n = cur->size();
+  for (size_t i = 0; i < n; i++) {
+    inheritRegisters(cur, aqlres.get(), _pos);
+    if (getOne(1, aqlres.get(), outputCounter)) {
+      outputCounter++;
```

Comment:
shouldn't outputcounter be modified in the function instead of increasing it depending on the function's result?